### PR TITLE
InvalidLinkBear: Deprecate timeout setting

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/coala/coala-bears/issues/1244

### CODE REVIEW

File: bears/general/InvalidLinkBear.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def find_links_in_file(file, network_timeout, link_ignore_regex,
                     yield line_number + 1, link, code
 
     @deprecate_settings(link_ignore_regex='ignore_regex')
+    @deprecate_settings(network_timeout='timeout')
```
Comment (sils) [inline bears/general/InvalidLinkBear.py]: just add it to the existing decorator, don't make two please

**Comment by sils** [comment]
Closes -> Fixes

**Comment by meetmangukiya** [comment]
But for a note, network_timeout isn't exactly timeout, it is a `dict`,
while timeout was `int`, so deprecation will not work actually...
Since the decorator passes the `timeout` setting as `network_timeout`,
which will make it malfunction, we need an error in this case and not
a deprecation warning.

**Comment by sils** [comment]
uhm deprecate_settings allows you to provide a conversion function, just read its docs, in this case you'll want a testcase for that as well. Maybe you want the dict to support something like `'*': 15` to match all URLs and then translate `15` to `{'*': 15}`

**Comment by sils** [comment]
we definitely need a testcase, 7902eb9 needs work

**Comment by meetmangukiya** [comment]
ahh,, My very first PR at coala was renaming `ignore_regex` to `link_ignore_regex`, but that was just a rename that's why the conversion didn't come into picture I guess, I made the assumption on that basis, I'll be looking into it now.

File: tests/general/InvalidLinkBearTest.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ def test_links_to_ignore(self):
         self.assertResult(valid_file=valid_file,
                           settings={'link_ignore_list': link_ignore_list})
 
-    def test_variable_timeouts(self):
+    def test_variable_timeouts_and_deprecation(self):
```
Comment (Asnelchristian) [inline tests/general/InvalidLinkBearTest.py]: `test_variable_timeouts` was okay  i think :sweat_smile:

**Comment by Asnelchristian** [comment]
this looks good to me 
ack https://github.com/coala/coala-bears/pull/1246/commits/c19e136b91849852228875b646338c6f12b50007

File: tests/general/InvalidLinkBearTest.py (hunk 2 above)
Comment (Mixih) [inline tests/general/InvalidLinkBearTest.py]: agreed

File: tests/general/InvalidLinkBearTest.py [hunk 3]
```diff
@@ -255,10 +255,19 @@ def response(status_code, *args, **kwargs):
             self.assertEqual([x.message
                               for x in list(uut.run('file', file_contents,
                                                     network_timeout=nt))], [])
+            self.assertEqual([x.message
+                              for x in list(uut.run('file', file_contents,
+                                                    timeout=20))], [])
```
Comment (Mixih) [inline tests/general/InvalidLinkBearTest.py]: well while we are at it, might as well add a test for the star syntax too.

**Comment by meetmangukiya** [comment]
@sils now does it look like what you requested earlier ?

**Comment by sils** [comment]
ack https://github.com/coala/coala-bears/pull/1246/commits/32a7d83177cbb98478b329d8b424d76030fff5ec

**Comment by sils** [comment]
remove WIP and rebase?

File: tests/general/InvalidLinkBearTest.py [hunk 4]
```diff
@@ -255,10 +257,26 @@ def response(status_code, *args, **kwargs):
             self.assertEqual([x.message
                               for x in list(uut.run('file', file_contents,
                                                     network_timeout=nt))], [])
+
+            with self.assertLogs(logging.getLogger()) as cm:
+                self.assertEqual([x.message
+                                  for x in list(uut.run('file', file_contents,
+                                                        timeout=20))], [])
+            self.assertEqual(cm.output,
```
Comment (Mixih) [inline tests/general/InvalidLinkBearTest.py]: hmm indent this please to line up so it's in the the with block. It's generally a bad idea to use a cm var outside of the with. (let me know if it's jsut github being stupid)

Comment (Mixih) [inline tests/general/InvalidLinkBearTest.py]: Also I'm not really a fan of the var name cm as it's not very descriptive. How about log?

**Comment by Mixih** [comment]
ack e6f8d32

**Comment by Mixih** [comment]
unack e6f8d32

**Comment by Mixih** [comment]
ack e20e679

**Comment by Mixih** [comment]
@rultor merge

**Pull request merged**